agent/azure-monitor-exporter/src/test/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorExportersEndToEndTest.java

```diff
@@ -0,0 +1,248 @@
+
+  @BeforeEach
+  public void setup() {
+    envVars.set(
+        "APPLICATIONINSIGHTS_CONNECTION_STRING", "InstrumentationKey=" + INSTRUMENTATION_KEY);
+  }
+
+  @Test
+  public void testBuildMetricExporter() throws Exception {
+    validateMetricExporterEndToEnd("testBuildMetricExporter");
+  }
+
+  @Test
+  public void testBuildTraceExporter() throws Exception {
+    validateTraceExporterEndToEnd("testBuildTraceExporter");
+  }
+
+  // OpenTelemetry doesn't have a Log API
+  @Test
+  public void testBuildLogExporter() throws Exception {
+    validateLogExporterEndToEnd();
+  }
+
+  @Test
+  public void testBuildTraceMetricLogExportersConsecutively() throws Exception {
+    validateTraceExporterEndToEnd("testBuildTraceMetricLogExportersConsecutively");
+    validateMetricExporterEndToEnd("testBuildTraceMetricLogExportersConsecutively");
+    validateLogExporterEndToEnd();
+  }
+
+  private static void validateMetricExporterEndToEnd(String testName) throws Exception {
+    CustomValidationPolicy customValidationPolicy = generateMetrics(testName);
+    TelemetryItem actualTelemetryItem = customValidationPolicy.getActualTelemetryItem();
+    TelemetryItem expectedTelemetryItem =
+        TestUtils.createAzureMonitorMetricTelemetry(
+            testName,
+            1,
+            INSTRUMENTATION_KEY,
+            actualTelemetryItem.getTime(),
+            actualTelemetryItem.getTags().get("ai.internal.sdkVersion"));
+    assertThat(actualTelemetryItem.getName()).isEqualTo(expectedTelemetryItem.getName());
+    assertThat(actualTelemetryItem.getInstrumentationKey())
+        .isEqualTo(expectedTelemetryItem.getInstrumentationKey());
+    assertThat(actualTelemetryItem.getTags()).isEqualTo(expectedTelemetryItem.getTags());
+    assertThat(actualTelemetryItem.getData().getBaseType())
+        .isEqualTo(expectedTelemetryItem.getData().getBaseType());
+    MetricsData expectedMetricsData = (MetricsData) expectedTelemetryItem.getData().getBaseData();
+    MetricsData actualMetricsData = (MetricsData) actualTelemetryItem.getData().getBaseData();
+    assertThat(actualMetricsData.getMetrics().get(0).getValue())
+        .isEqualTo(expectedMetricsData.getMetrics().get(0).getValue());
+    assertThat(actualMetricsData.getAdditionalProperties())
+        .isEqualTo(expectedMetricsData.getAdditionalProperties());
+  }
+
+  private static void validateTraceExporterEndToEnd(String testName) throws Exception {
+    CustomValidationPolicy customValidationPolicy = generateTraces(testName);
+    TelemetryItem actualTelemetryItem = customValidationPolicy.getActualTelemetryItem();
+    TelemetryItem expectedTelemetryItem =
+        TestUtils.createAzureMonitorRemoteDependencyTelemetry(
+            testName,
+            INSTRUMENTATION_KEY,
+            actualTelemetryItem.getTime(),
+            actualTelemetryItem.getTags().get(ContextTagKeys.AI_OPERATION_ID.toString()),
+            actualTelemetryItem.getTags().get("ai.internal.sdkVersion"));
```

Comment:
same here

Reply:
whenever we upgrade opentelemetry and java version, it will fail. it's in this format "java11.0.10:otel1.15.0:ext1.0.0-beta.4". 
I think it's better to use actual telemetry item's value.

Reply:
you can use `assertThat(...).matches(...)` instead of hard-coded value so it won't break on upgrade